### ISSUE TITLE
The ability for UX canvases to curve similar to the curve plane mesh

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1443]
### Is your feature request related to a problem? Please describe.

Currently if you want to have a curved UX canvas you need to render it using a camera and then put the render output onto a curved plane mesh then you need to use components to make sure they're laser interactions work properly

### Describe the solution you'd like

A native solution for the canvas itself to be curved this increases clarity of the UI along with it being more optimized since there's no camera being used

### Describe alternatives you've considered

Already described the alternatives in the first box

### Additional Context

Now the reason why this might not be added is because UIX simply may not work this way at all and might be impossible without a complete rewrite of the UI rendering system

### Requesters

@epicEaston197

**Comment by epicEaston197** [comment]
>I'm not certain I understand what is meant by 'clarity of the UI'

right now the UIX for things like the dash are on a hidden layer being rendered by a camera with a fixed resolution the camera's output is then placed onto the curved plane mesh that is our dash this results in a lower quality look of the UI due to the cameras output if you just looked at the plane UIX itself it would be much higher quality similar to the inspectors

the quality difference isn't that drastic though but would increase the ease ability of creating curved UIX

**Comment by Frooxius** [comment]
I don't think this is something we would implement, because of the significant complexity and performance impacts of this.

The native UIX cannot be curved easily, because it's all geometry. A lot of it can be pretty big pieces of geometry - e.g. a large quad that's the entire background of a panel.

You cannot just curve this, because there's no inner geometry to curve - which means that we would need to split the triangles and generate additional geometry for the curve. This would end up increasing the computational and geometric complexity and making sure it all stays properly aligned with each other.

This gets even more complex the more types of curving you want to support as well - essentially each triangle would need to be able to be cut into a grid, so the vertices can then be projected.

Using texture is the easiest and probably most performant solution you can have for curved UI's, so I don't think we'd invest time into making this happen in geometry.

**Comment by epicEaston197** [comment]
That is understandable that's kinda what I was expecting but really interesting to hear your reasoning nonetheless

**Comment by TisFoolish** [comment]
What about a compliment that makes projecting UIX to one of the primitive meshes easier? Eg a sort of "UIXTexture"?

**Comment by Frooxius** [comment]
We could add things to make it easier to setup the sizing/positioning of the rendering, but I think that's a bit of a different issue from this.

**Comment by epicEaston197** [comment]
> What about a compliment that makes projecting UIX to one of the primitive meshes easier? Eg a sort of "UIXTexture"?

okay I made issue #1446 around this topic please add additional context or recommendations if you would like to
